msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/ADAP3DFeatureDetectionMethod.java

```diff
@@ -149,4 +188,22 @@
     return featureList;
 
   }
+
+  /** {@inheritDoc} */
+  @Override
+  public Float getFinishedPercentage() {
+    return objPeakDetection.getFinishedPercent();
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public List<Feature> getResult() {
+    return finalFeatureList;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public void cancel() {
+    objPeakDetection.cancel();
```

Comment:
Also, you should check for the cancellation flag in the ADAP3DFeatureDetectionMethod.execute() method - remember the code flow will continue after objPeakDetection.execute() if objPeakDetection.cancel() is called.